### ISSUE TITLE
[feature-request] default-priors for random effects SDs

### ISSUE THREAD

**Issue by GidonFrischkorn** [popov-lab/bmm#342]
### Problem

bmm currently relies on the brms default `student_t(3, 0, 2.5)` for all random effects standard deviations. This is far too wide for most bmm parameters, which operate on constrained scales:

| Parameter | Link | What SD = 2.5 implies |
|---|---|---|
| kappa, c, a, s | log | Individual values vary by up to exp(2.5) ≈ 12× from group mean |
| thetat, thetant | identity (logit scale) | Individual mixing weights span nearly full [0, 1] range |
| mu | tan_half | Individual biases span the entire circle |

These priors are not domain-informed and can lead to slow sampling or poorly identified models, especially with small sample sizes.

### Proposal

Add parameter-type-specific SD priors to model definitions, similar to how `default_priors` already distinguishes `main` and `effects` priors for fixed effects. For example:

| Parameter type | Link | Proposed SD prior | Interpretation |
|---|---|---|---|
| kappa, c, a, s | log | `exponential(1)` | ~68% of individuals within ×0.37 to ×2.72 of group mean |
| thetat, thetant | identity | `exponential(1)` | SD of 1 on logit scale; moderate individual differences |
| mu/mu1 | tan_half | `exponential(1)` | Rarely relevant (mu usually fixed to 0) |

### Implementation

- Extend model constructors with an `sd_priors` field (or add an `sd` key to `default_priors`)
- Extend `set_default_prior()` / `configure_prior.bmmodel()` to detect random effects in the formula and apply `class = "sd"` priors
- Add test coverage for SD priors across different random effects structures (intercept-only, slopes, crossed)

### Context

Discussed in #306. Related to the broader effort of making bmm's default priors domain-informed rather than inheriting generic brms defaults.

**Comment by pmusfeld** [comment]
I was just having a similar thought yesterday, when adjusting hyperpriors on standard deviation parameters in the M3 model. It's of course already possible by just adding custom priors when calling the fit-function but I though it would be convenient to be able to adjust priors on standard deviation over the `default_prior` argument in the model specification. 

I think it would be really convent if this feature could be added! Maybe this could be build analogous to the `main` and the `effects` argument currently used for setting priors on intercept and effect coefficients? for example:

```
m3_model <- m3(
    resp_cats = c("target", "withinList", "extraList"),
    num_options = c(1, 2, 3),
    choice_rule = "softmax",
    links = list(
      a = "log",
      c = "log"
    ),
    default_priors = list(
      a = list(
        main = "normal(0, 0.7)", 
        effects = "normal(0, 0.5)", 
        sd_main = "exponential(1)", 
        sd_effects = "exponential(1)"
      ),
      c = list(
        main = "normal(0, 7)", 
        effects = "normal(0, 0.5)", 
        sd_main = "exponential(1)", 
        sd_effects = "exponential(1)"
      )
   )
)
```

**Comment by GidonFrischkorn** [comment]
## Implementation Plan: Default Priors for Random Effects SDs

Based on the PriorsBinomial workflow (Frischkorn, Dutli, Musfeld & Oberauer), prior **scale** matters far more than family for SD priors. The `exponential` family is recommended, with rate calibrated to each parameter's link scale.

### API: Extend `default_priors` with `sd_main` / `sd_effects`

```r
kappa = list(main = "normal(2, 1)", effects = "normal(0, 1)",
             sd_main = "exponential(1)", sd_effects = "exponential(1)")
```

- `sd_main` → prior on SD of random intercepts (`class="sd", coef="Intercept"`)
- `sd_effects` → blanket prior on SD of random slopes (`class="sd"`)
- Both keys always paired together
- brms priority rules: specific intercept prior overrides blanket for intercept; blanket covers slopes

### Prior Selection Rule

Rate = 1/prior_SD, rounded to clean values. Replaces brms' generic `student_t(3, 0, 2.5)`.

| Model family | Parameters | sd_main | sd_effects |
|---|---|---|---|
| Circular (mixture2p/3p, SDM, IMM) | kappa, c, a, s, thetat, thetant | exponential(1) | exponential(1) |
| EZDM | drift | exponential(1) | exponential(2) |
| EZDM | bound | exponential(2) | exponential(2) |
| EZDM | ndt, zr | exponential(2) | exponential(3) |
| M3 softmax | a | exponential(1) | exponential(2) |
| M3 softmax | c, f | exponential(1) | exponential(1) |
| M3 simple | a, c | exponential(2) | exponential(2) |
| M3 simple | f | exponential(1) | exponential(1) |

### Implementation (4 commits)

1. **Prior infrastructure** — new helpers in `R/helpers-prior.R`: `.build_sd_prior()`, `construct_sd_priors()`, `.extract_re_terms()`, extend `construct_default_priors_list()`
2. **Model constructors** — add `sd_main`/`sd_effects` to all 6 model files
3. **Tests** — intercept-only, intercept+slope, slope-only, no RE, user override, multiple groups, mock fits
4. **Docs** — update `summarise_default_prior()`, roxygen, design doc

**Comment by GidonFrischkorn** [comment]
After implementing the `sd_main`/`sd_effects` split proposed here, I reconsidered whether that distinction is actually necessary given how brms resolves SD priors.

brms uses a specificity hierarchy: a prior with `coef = "Intercept"` and `group = "ID"` overrides a blanket prior with no `coef` or `group`. The original design exploited this by setting `sd_main` as an Intercept-specific prior and `sd_effects` as a blanket, so intercept SDs would get `sd_main` and slope SDs would get `sd_effects`.

In practice:

- **users use random intercepts only** `(1 | ID)`. In that case `sd_effects` (the blanket) is never reached, there are no slope SDs to cover. Only `sd_main` ever matters.
- **14 of 20 parameter definitions had `sd_main == sd_effects` anyway.** For these, emitting a redundant Intercept-specific prior with the same value as the blanket adds noise to the prior table without changing any behaviour.
- **6 cases where they differed** (ezdm `drift`/`ndt`/`zr`, m3 softmax `a`), the distinction only does anything when the user has random slopes. Users writing `(1 + condition | ID)` are already making a deliberate modelling decision and can specify slope SD priors themselves via the `prior` argument to `bmm()`.

I instead implemented a single `sd` key per parameter in `default_priors`, set as a global blanket (no `coef`, no `group`). This:

- Replaces the brms default `student_t(3, 0, 2.5)` for all SD parameters of a given model parameter, regardless of whether they are intercept or slope SDs
- Keeps the model constructors and prior-building code simple
- Preserves full user control: any call to `bmm(..., prior = brms::prior_("...", class = "sd", coef = "Intercept", group = "ID", nlpar = "kappa"))` overrides the blanket for that specific SD

Users who genuinely need separate priors for intercept vs. slope SDs can supply them directly, I think that use case is specific enough that it shouldn't be baked into the defaults.
